taigit/src/libraries/Taiga.ts

```diff
@@ -498,4 +498,114 @@ taiga_issues(projId : number) : Promise<Object>{
+ *          fin_per : number,     //estimated task finished percentage
+ *          detail: Array<Object>  //detailed log of task status change
+ *             The Object in the Array has the following data type:
+ *             <Object>：
+ *               state_trans_valid: true,
+ *               timecost: number,  //time of history entry cost in milliseconds
+ *               acctime: number,   //time of task accumulated of this history entry cost in milliseconds
+ *               status_trans: String[],//Status transition array
+ *               user: object,         // Taiga User Object
+ *               date: number,        //Date and time of history entry in milliseconds since epoch
+ * }
+ */
+export async function
+task_assessment(taskId : number) : Promise<Object> {
+    let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+    let output : Array<Object> = [];
+    let pre : number = 0;
+    let start : number = 0;
+    let endstatus : string = "New";
+    let task_valid : boolean = true;
+    for(let entry of data) {
+        if(entry.values_diff.status) {//use this to ignore no-status change diff
+            let new_entry = {
+                state_trans_valid: true,
+                timecost: 0,
+                acctime: 0,
+                status_trans: entry.values_diff.status,
+                user: entry.user,
+                date: new Date(entry.created_at).getTime(),
+            }
+            endstatus = new_entry.status_trans[1];
+            let timecost: number;
+            let acctime: number;
+            if (new_entry.status_trans[0] == "New") {
+                pre = new_entry.date;
+                acctime = 0;
+                new_entry.acctime = acctime;
+                start = new_entry.date;
+            } else {
+                timecost= new_entry.date - pre;
+                acctime = new_entry.date - start;
+                new_entry.timecost = timecost;
+                new_entry.acctime = acctime;
+            }
+            let old_status = new_entry.status_trans[0];
+            let new_status = new_entry.status_trans[1];
+            if ((old_status == "New" && new_status != "In progress") ||
+                (old_status == "In progress" && new_status != "Ready for test") ||
+                (old_status == "Ready for test" && new_status != "Closed")) {
+                new_entry.state_trans_valid = false;
+                task_valid = false;
+            }
+        output.push(new_entry);
+        }
+    }
+    let t_fin_per : number;
+    let t_num_stat : number;
+    let t_finished : boolean = false;
+    if(endstatus == "Closed") {
+        t_finished = true;
+    }
+    //transfer ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    //into number status
+    switch(endstatus) {
+        case "Closed": {
+            t_fin_per = 100;
```

Comment:
Do we still need these (t_fin_per) values if all that's being returned is the transitions and whether or not the overall transition tracking was valid?

Reply:
task finish percentage is an supplement way to show the burndown chart in task-wise accuracy.

Reply:
Frontend from what I recall is not making a component that shows this information. @trevorforrey can elaborate/confirm. He's looking for user story history and that's it.